src/visualizer.jl

```diff
@@ -0,0 +1,71 @@
+struct MechanismVisualizer{M <: MechanismState}
+    state::M
+    visualizer::Visualizer
+    modcount::Int
+
+    function MechanismVisualizer(state::M,
+                                 frame_to_visuals::Associative{CartesianFrame3D}=create_visuals(state.mechanism),
+                                 vis::Visualizer=Visualizer()) where {M <: MechanismState}
+        mvis = new{M}(state, vis, rbd.modcount(state.mechanism))
+        _set_mechanism!(mvis, frame_to_visuals)
+        _render_state!(mvis)
+        mvis
+    end
+end
+
+MechanismVisualizer(m::Mechanism, args...) = MechanismVisualizer(MechanismState{Float64}(m), args...)
+MechanismVisualizer(m::Mechanism, fname::AbstractString, args...; kw...) =
+    MechanismVisualizer(m, parse_urdf_visuals(fname, m; kw...), args...)
+MechanismVisualizer(m::MechanismState, fname::AbstractString, args...; kw...) =
+    MechanismVisualizer(m, parse_urdf_visuals(fname, m.mechanism; kw...), args...)
+
+to_affine_map(tform::Transform3D) = AffineMap(rotation(tform), translation(tform))
+
+function _set_mechanism!(mvis::MechanismVisualizer, frame_to_visuals)
+    mechanism = mvis.state.mechanism
+    vis = mvis.visualizer
+    tree = mechanism.tree # TODO: tree accessor?
+    for vertex in rbd.Graphs.vertices(tree)
+        body = vertex
+        body_ancestors = rbd.Graphs.ancestors(vertex, tree)
```

Comment:
Cool, glad to see the `RBD.Graphs` stuff coming in handy.

Reply:
Yup, it's great!